modules/cddl-gen/CMakeLists.txt

```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+if(NOT CONFIG_CDDL_GEN)
+  return()
+endif()
+
+add_subdirectory(${ZEPHYR_CDDL_GEN_MODULE_DIR} cddl-gen)
```

Comment:
Why not do:
```suggestion
add_subdirectory_ifdef(CONFIG_CDDL_GEN ${ZEPHYR_CDDL_GEN_MODULE_DIR} cddl-gen)
```

Reply:
haha, thank you! That can be considered a brainfart

Reply:
never mind
I didn't notice you were adding `cddl-gen` as binary dir.

Reply:
Discussed offline, `add_subdirectyr_ifdef` does not pass build dir argument. So brainfart remains.